leftsidebar: Use the same @ symbol everywhere.

Rinwaoluwa opened the pull request:
This replaces the font awesome @ icon and @ character with custom feather icon @ sign for "Mentions" navigations label, unread mention's indicator in the left sidebar, User popovers "Copy mention syntax" & "Reply mention syntax".

Fixes #22816

Co-authored-by: Lai Ying Wong


**Screenshots and screen captures:**
**MENTIONS**
| Before | After |
|---|---|
| <pre>![BeforeLeftside@Light](https://github.com/zulip/zulip/assets/117735533/e0880f52-c1a2-4663-8b5f-7d951207f057)</pre> | <pre>![AfterLeftside@Light](https://github.com/zulip/zulip/assets/117735533/688d7e98-cb5d-4398-8eec-9e5e211a6335)</pre> |
| <pre>![BeforeLeftside@Dark](https://github.com/zulip/zulip/assets/117735533/3e13dd86-83b6-494c-8ccf-351d694df607)</pre> | <pre>![AfterLeftside@Dark](https://github.com/zulip/zulip/assets/117735533/913189c6-2dfc-421a-90cd-37240d715251)</pre> |


**USER POPOVERS "Copy mention syntax"**
| Before | After |
|---|---|
| <pre>![BeforePopoverCopy@Lightpng](https://github.com/zulip/zulip/assets/117735533/ba2ea183-17f8-42a3-ae5d-cc7fdce62f9c)</pre> | <pre>![AfterPopoverCopy@Light](https://github.com/zulip/zulip/assets/117735533/cda9dbd2-6d97-40e3-81d4-d51c14ffe67d)</pre> |
| <pre>![BeforePopoverCopy@Dark](https://github.com/zulip/zulip/assets/117735533/b20be6b0-e1f2-46b7-8224-1a7f1a879b11)</pre> | <pre>![AfterPopoverCopy@Dark](https://github.com/zulip/zulip/assets/117735533/2a866627-d72c-41aa-8801-97de740cc45d)
</pre> |

**(new) Unread mentions indicator in the left sidebar LIGHT THEME**

https://github.com/zulip/zulip/assets/117735533/bc465656-a57f-4c4d-aadf-6ee3ef67fc99


https://github.com/zulip/zulip/assets/117735533/08a83eb5-cb05-4172-8b38-310b9fd25aed




<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

Rinwaoluwa commented:
@alya, would you mind reviewing this ?

alya commented:
Thanks! The screenshots look reasonable to me. 

@karlstolley Are you up for reviewing this one? I haven't tested.

Rinwaoluwa commented:
@karlstolley your review is needed here as well please.

karlstolley commented on web/tests/stream_list.test.js:
```diff
@@ -102,7 +102,7 @@ function create_social_sidebar_row({mock_template}) {
     stream_list.create_sidebar_row(social);
     assert.equal($social_count.text(), "99");
-    assert.equal($social_unread_mention_info.text(), "@");
+    assert.equal($social_unread_mention_info.text(), "never-been-set");
```
These assertions don't make sense. There needs to be a check that the new icon representation is visible.

karlstolley commented on web/styles/settings.css:
```diff
@@ -639,6 +639,11 @@ input[type="checkbox"] {
 
+.zulip-icon-at-sign {
+    position: relative;
+    top: 3px;
```
I'm not a fan of these kinds of fiddly, position-based adjustments. It would be useful to look into making a parent container around the icon and label, and using a modern technique like flexbox to get the alignment to work.

Rinwaoluwa replied:
I've update the PR with this changes

- Add a div container inside th cell with a class of notification-table-header-content.
- Set baseline alignment on the group, and center alignment on the icons on a per-flex-item basis.
- Added a new container for at-sign and all text to center them vertically;

Rinwaoluwa commented:
@karlstolley This still needs a review please.

alya commented:
@Rinwaoluwa If you have addressed feedback from @karlstolley 's prior review, please post a comment explaining how you did so.

alya commented:
To a reviewer, this PR does not appear to be ready for another review pass.

Rinwaoluwa commented on web/tests/stream_list.test.js:
```diff
@@ -79,7 +79,7 @@ function create_devel_sidebar_row({mock_template}) {
     stream_list.create_sidebar_row(devel);
     assert.equal($devel_count.text(), "42");
-    assert.equal($devel_unread_mention_info.text(), "");
+    assert.equal($devel_unread_mention_info.is(":visible"), false);
```
@karlstolley I've update the PR to Test for the Icon's visibility right here within the test `$devel_unread_mention_info` represents the `@` icon for an unread mention

karlstolley commented:
You've still not addressed all of the feedback I've left. And it doesn't make sense to have one commit that says a specific area for where the `@` symbol has been changed, followed by another that says "everywhere."

Rinwaoluwa commented:
@karlstolley  Thank you for taking the time to review my pull request. I appreciate your feedback about reordering the commit history. I've update the PR according to your previous reviews as referenced in [CZO](https://chat.zulip.org/#narrow/stream/91-code-review/topic/Use.20the.20same.20.40.20symbol.20everywhere/near/1739828)
**Changes**
- Added a div container inside th cell with a class of notification-table-header-content.
- Set baseline alignment on the group (notification-table-header-content), and center alignment on the icons on a per-flex-item basis.
- Added a new container for at-sign and all text to center them vertically;

Could you please clarify any other outstanding feedback you have on the PR that I haven't addressed yet?

karlstolley commented:
A commit that says "everywhere" is still misleading. It's not "everywhere." Please be precise in your wording and your work, and imagine a time where some future contributor might need to go back through the commit history. If they find your commit with "everywhere" in it, you're misleading them.

Rinwaoluwa commented:
@karlstolley Thanks for pointing that out! You're absolutely right, "everywhere" is not very specific. I've updated the commit message to be more precise.

karlstolley commented on web/templates/settings/notification_settings.hbs:
```diff
@@ -8,22 +8,27 @@
                 <tr>
-                    <th rowspan="2" width="30%"></th>
-                    <th colspan="2" width="28%">{{t "Desktop"}}</th>
+                    <th rowspan="2" width="30%"><div class="notification-table-header-content"></div></th>
```
I don't understand the inclusion of this element everywhere, especially when it's empty.

AMANVISHWAKARMA27 commented:
Hey @Rinwaoluwa are you still working on this ? I would like to take up the issue and continue from where you left off, but if you still are working on this no worries I'll pick up something else.

Rinwaoluwa commented:
@AMANVISHWAKARMA27 No please take It up and have fun while your at it. Wish you the best.

AMANVISHWAKARMA27 commented:
> @AMANVISHWAKARMA27 No please take It up and have fun while you're at it. Wish you the best.

Thanks a lot @Rinwaoluwa , I will co-author you in my commits.
